documentation/sign-in-canada.md

```diff
@@ -0,0 +1,36 @@
+# Sign In Canada
+
+## Local setup
+
+### Environment variables
+
+#### `api/.env`
+
+- Comment out block of variables 'for mock oauth testing'
+- Uncomment block of variables 'for SiC testing'
+- Change `OAUTH_ACR_VALUES=gckeymfa` to `OAUTH_ACR_VALUES=gckey`
+- Add values for `OAUTH_API_CLIENT_ID` and `OAUTH_API_CLIENT_SECRET` (these can be obtained from @tristan-orourke)
+
+#### `apps/web/.env`
+
+- Comment out block of variables 'Mock auth endpoint'
+- Uncomment block of variables 'SiC endpoint (will break e2e tests)'
+- Change `OAUTH_ACR_VALUES=gckeymfa` to `OAUTH_ACR_VALUES=gckey`
+- Run `npm run dev` to update config values
+
+### Account creation
+
+1. Navigate to [Sign in using GCkey](http://localhost:8000/en/login-info)
+2. Click the _Continue to GCKey and sign in_ button
+3. Sign in or Sign up for an account
+4. Start your account at the Create an account page
+5. Copy the email address entered in Create an account page form for later
+6. Run `php artisan tinker`
+7. Run the following code block in tinker to add `base_user`, `applicant`, and `platform_admin` roles to the user previously created (replace *username@domain.tld* with email previously copied in step 5)
+
+```
+$user = User::where('email', 'username@domain.tld')->sole();
+$roles = ['base_user', 'applicant', 'platform_admin'];
+$user->attachRoles($roles)
```

Comment:
I played around with this a bit.  The command is actually "addRoles", not "attachRoles".  We tend to use syncRoles in our app but addRoles works just as well here.  You should probably use only one or other, though.  Both is redundant, I think.  And if you use only one, you probably can use the roles array inline with the function call instead of assigning it to the variable `$roles` first.

Reply:
Reading is fundamental. I was looking at Laratrust **version 6** [documentation](https://laratrust.santigarcor.me/docs/6.x/usage/roles-and-permissions.html#user-roles-assignment-removal) 🤕.

Updated in adbe7c06fe7a9c4771d95e167e6ff360b6df0d4a.